### PR TITLE
Lista Distributor integration

### PR DESCRIPTION
**Pull request by lawson-ccy**
The PR is to implement the integration of Lista Distributors. User are able to earn rewards by borrow LisUSD from their collaterals and stakes LisUSD.

1. Added interfaces for 2 new Lista Distributor contract.
2. Integrated Lista Distributor at the the `borrow` and `payback` function the `Interaction` contract.
3. Integrated Lista Distributor at the the `join`(stake) and `exit`(unstake) function the `Jar` contract.

Also, it contains a minor amendment of deprecating the usage of `HelioRewards`, related codes has been removed in this PR.

### CODE REVIEW

File: contracts/Interaction.sol [hunk 1]
```diff
@@ -77,6 +80,11 @@ contract Interaction is OwnableUpgradeable, IDao, IAuctionProxy {
         for(uint256 i = 0; i < tokens.length; i++)
             tokensBlacklist[tokens[i]] = 0;
     }
+    function setListaDistributor(address distributor) external auth {
+        require(distributor != address(0), "Interaction/lista-distributor-zero-address");
+        require(address(borrowLisUSDListaDistributor) == distributor, "Interaction/same-distributor-address");
```
Comment (qingyang-lista) [inline contracts/Interaction.sol:85]: should be `!=`

File: contracts/jar.sol [hunk 2]
```diff
@@ -183,6 +185,11 @@ contract Jar is Initializable, ReentrancyGuardUpgradeable {
         live = 1;
         emit Uncage();
     }
+    function setListaDistributor(address distributor) external auth {
+        require(distributor != address(0), "Interaction/lista-distributor-zero-address");
+        require(address(stakeLisUSDListaDistributor) == distributor, "Interaction/same-distributor-address");
```
Comment (qingyang-lista) [inline contracts/jar.sol:190]: should be `!=`

File: contracts/Interaction.sol (hunk 1 above)
Comment (lawson-ccy) [reply contracts/Interaction.sol]: Fixed

File: contracts/jar.sol (hunk 2 above)
Comment (lawson-ccy) [reply contracts/jar.sol]: Fixed

**Comment by Khaliq88** [comment]
> The PR is to implement the integration of Lista Distributors. User are able to earn rewards by borrow LisUSD from their collaterals and stakes LisUSD.
> 
> 1. Added interfaces for 2 new Lista Distributor contract.
> 2. Integrated Lista Distributor at the the `borrow` and `payback` function the `Interaction` contract.
> 3. Integrated Lista Distributor at the the `join`(stake) and `exit`(unstake) function the `Jar` contract.
> 
> Also, it contains a minor amendment of deprecating the usage of `HelioRewards`, related codes has been removed in this PR.

repo-sync

**Review by qingyang-lista — APPROVED** [review]
LGTM

**Pull request merged**